vedant3620/day25/middle_element.cpp

```diff
@@ -0,0 +1,79 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include<iostream>
+using namespace std;
+
+/* Link list Node */
+struct Node
+{
+    int data;
+    struct Node* next;
+    
+    Node(int x){
+        data = x;
+        next = NULL;
+    }
+};
+
+void append(struct Node** head_ref, struct Node **tail_ref, int new_data)
+{
+    struct Node* new_node = new Node(new_data);
+    
+    if (*head_ref == NULL)
+       *head_ref = new_node;
+    else
+       (*tail_ref)->next = new_node;
+    *tail_ref = new_node;
+}
+
+
+/* Function to get the middle of the linked list*/
+int getMiddle(struct Node *head);
+
+/* Driver program to test above function*/
+int main()
+{
+  int T,i,n,l;
+
+    cin>>T;
+
+    while(T--){
+    struct Node *head = NULL,  *tail = NULL;
+
+        cin>>n;
+        for(i=1;i<=n;i++)
+        {
+            cin>>l;
+            append(&head, &tail, l);
+        }
+
+    printf("%d\n", getMiddle(head));
+    }
+    return 0;
+}
+// } Driver Code Ends
+/* Link list Node 
+struct Node {
+    int data;
+    Node* next;
+    
+    Node(int x){
+        data = x;
+        next = NULL;
+    }
+    
+}; */
+
```

Comment:
Remember to remove unused commented code from the files. Explanation comments are okay.